File: app/src/main/java/com/demo/railbridge/sdk/ScenarioPreset.java
package com.demo.railbridge.sdk;

import java.util.Locale;

public enum ScenarioPreset {
    NORMAL("normal"),
    TIMEOUT("timeout"),
    INTERNAL_ERROR("internal_error"),
    CALLBACK_LOSS("callback_loss"),
    DUPLICATE_CALLBACK("duplicate_callback"),
    RETRY_EXHAUSTED("retry_exhausted");

    private final String value;

    ScenarioPreset(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public static ScenarioPreset fromValue(String rawValue) {
        if (rawValue == null || rawValue.trim().isEmpty()) {
            return NORMAL;
        }

        String normalized = rawValue.trim()
                .toLowerCase(Locale.US)
                .replace('-', '_')
                .replace(' ', '_');

        for (ScenarioPreset preset : values()) {
            if (preset.value.equals(normalized) || preset.name().toLowerCase(Locale.US).equals(normalized)) {
                return preset;
            }
        }

        return NORMAL;
    }
}

File: ios/RailBridgeIOS/RailBridgeIOS/SDK/ScenarioPreset.swift
import Foundation

enum ScenarioPreset: String, CaseIterable, Codable {
    case normal = "normal"
    case timeout = "timeout"
    case internalError = "internal_error"
    case callbackLoss = "callback_loss"
    case duplicateCallback = "duplicate_callback"
    case retryExhausted = "retry_exhausted"

    static func from(_ rawValue: String?) -> ScenarioPreset {
        guard let rawValue else {
            return .normal
        }

        let normalized = rawValue
            .trimmingCharacters(in: .whitespacesAndNewlines)
            .lowercased()
            .replacingOccurrences(of: "-", with: "_")
            .replacingOccurrences(of: " ", with: "_")

        return ScenarioPreset(rawValue: normalized) ?? .normal
    }

    static var allValues: [String] {
        allCases.map(\.rawValue)
    }
}
